Sources/WalletConnectNotify/Client/Wallet/NotifyClient.swift

```diff
@@ -106,6 +106,7 @@ public class NotifyClient {
         try await identityClient.unregister(account: account)
         notifyWatcherAgreementKeysProvider.removeAgreement(account: account)
         try notifyStorage.clearDatabase(account: account)
+        Task { try await pushClient.unregister()}
```

Comment:
What if the unregister fails (e.g. no internet?) Will this flow be tried again or will the push registration still be present?

Reply:
good point, I will make it throw

Reply:
@chris13524 what code push api returns for already unregistered client on /unregister ?

Reply:
It gives the same response code regardless if the client was already unregistered or not: 200

Reply:
> I will make it throw

I still think there is an issue. If it throws, what does caller do with that? If simply display an error message, the question to me is when will it do the unregister again?

Looks to me that you are clearing the notify storage database (and other things) before you unregister the push client. So this would mean if the way the unregister takes place is based on the notify storage, then the unregister won't happen again. I'm assuming that logout should be an atomic operation to prevent the receiving of push notifications it should not.